backend/controllers/cellMove/selectLocation.js

```diff
@@ -1,18 +1,30 @@
-const moment = require('moment')
-
 const { serviceUnavailableMessage } = require('../../common-messages')
 const { alertFlagLabels, cellMoveAlertCodes } = require('../../shared/alertFlagValues')
 const { putLastNameFirst } = require('../../utils')
+const { showNonAssociationsLink, showCsraLink } = require('./cellMoveUtils')
 const {
   app: { notmEndpointUrl: dpsUrl },
 } = require('../../config')
 
-module.exports = ({ elite2Api, logError }) => async (req, res) => {
+module.exports = ({ elite2Api, whereaboutsApi, logError }) => async (req, res) => {
   const { offenderNo } = req.params
+  const { missingLocation } = req.query
+  const errors = []
+  if (missingLocation) {
+    errors.push({ text: 'Select residential unit', href: '#location' })
+  }
 
   try {
     const prisonerDetails = await elite2Api.getDetails(res.locals, offenderNo, true)
     const nonAssociations = await elite2Api.getNonAssociations(res.locals, prisonerDetails.bookingId)
+    const cellAttributesData = await elite2Api.getCellAttributes(res.locals)
+    const locationsData = await whereaboutsApi.searchGroups(res.locals, prisonerDetails.agencyId)
+
+    const locations = locationsData.map(location => ({ text: location.name, value: location.key }))
+    locations.unshift({ text: 'All locations', value: 'ALL' })
+    const cellAttributes = cellAttributesData
+      .filter(cellAttribute => 'Y'.includes(cellAttribute.activeFlag))
```

Comment:
Could we just do `cellAttribute.activeFlag === 'Y'`?

Reply:
Yeah, I think this may be my Java side coming and Matt C advising that was preferable.

Reply:
Will fix in the next ticket.